__tests__/distributors/zulu.test.ts

```diff
@@ -0,0 +1,525 @@
+    await expect(
+      zuluDistributorPrototype.getAvailableVersions.call(zuluDistributor)
+    ).resolves.not.toBeNull();
+  });
+
+  it('Error is thrown for empty response', async () => {
+    spyHttpClient.mockImplementationOnce(
+      async (): Promise<ifm.ITypedResponse<IZuluVersions[]>> => {
+        return {
+          statusCode: 200,
+          headers: {},
+          result: [] as IZuluVersions[]
+        };
+      }
+    );
+    zuluDistributor = new ZuluDistributor({ version: '11', arch: 'x86', packageType: 'jdk' });
+    const zuluDistributorPrototype = Object.getPrototypeOf(zuluDistributor);
+    await expect(
+      zuluDistributorPrototype.getAvailableVersions.call(zuluDistributor)
+    ).rejects.toThrowError(/No versions were found using url */);
+  });
+
+  it('Error is thrown for undefined', async () => {
+    spyHttpClient.mockImplementationOnce(
+      async (): Promise<ifm.ITypedResponse<IZuluVersions[] | undefined>> => {
+        return {
+          statusCode: 200,
+          headers: {},
+          result: undefined
+        };
+      }
+    );
+    zuluDistributor = new ZuluDistributor({ version: '11', arch: 'x86', packageType: 'jdk' });
+    const zuluDistributorPrototype = Object.getPrototypeOf(zuluDistributor);
+    await expect(
+      zuluDistributorPrototype.getAvailableVersions.call(zuluDistributor)
+    ).rejects.toThrowError(/No versions were found using url */);
+  });
+});
+
+describe('getArchitectureOptions', () => {
+  let zuluDistributor = new ZuluDistributor({ version: '11', arch: 'x86', packageType: 'jdk' });
+  const zuluDistributorPrototype = Object.getPrototypeOf(zuluDistributor);
+
+  it.each([
+    [{ architecture: 'x64' }, { arch: 'x86', hw_bitness: '64', abi: '' }],
+    [{ architecture: 'x86' }, { arch: 'x86', hw_bitness: '32', abi: '' }],
+    [{ architecture: 'x32' }, { arch: 'x32', hw_bitness: '', abi: '' }]
+  ])('%o -> %o', (input, expected) => {
+    expect(zuluDistributorPrototype.getArchitectureOptions.call(input)).toEqual(expected);
+  });
+});
+
+describe('findPackageForDownload', () => {
+  let spyHttpClient: jest.SpyInstance;
+  let zuluDistributor: ZuluDistributor;
+
+  beforeEach(() => {
+    spyHttpClient = jest.spyOn(HttpClient.prototype, 'getJson');
+    spyHttpClient.mockImplementation(
+      async (url): Promise<ifm.ITypedResponse<IZuluVersions[]>> => {
+        manifestData = require('../data/zulu/zulu-releases-default.json');
+        if (url.includes('javafx=true')) {
+          manifestData = require('../data/zulu/zulu-releases-fx.json');
+        }
+
+        if (url.includes('release_status=ea')) {
+          manifestData = require('../data/zulu/zulu-releases-ea.json');
+        }
+
+        if (url.includes('bundle_type=jdk')) {
+          manifestData = manifestData.filter((item: any) => item.name.includes('-jdk'));
+        } else {
+          manifestData = manifestData.filter((item: any) => item.name.includes('-jre'));
+        }
```

Comment:
Why do we need so many json files and so much logic?
What is the difference between `zulu-releases-default.json`, `zulu-releases-fx.json`, `zulu-releases-ea.json`?
I think they are almost identical so we don't need all of them

Reply:
Done. Thank you.